src/pages/ReservationsPage.js

```diff
@@ -1,45 +1,41 @@
+// import { useEffect } from 'react';
 import { useSelector, shallowEqual } from 'react-redux';
+import { Navigate } from 'react-router-dom';
+import { getReserv } from '../redux/reserv/reserv';
+import store from '../redux/configureStore';
 
 function ReservationsPage() {
-  const reservations = useSelector((state) => state.reservations.reservations, shallowEqual);
-  console.log(reservations);
-  let codefragment = null;
-  if (reservations === undefined || reservations === []) {
-    codefragment = (
-      <>
-        <h1>Reservations Page</h1>
-        <ul className="reservtable">
-          <li className="listitem">
-            <span>Car</span>
-            <span>City</span>
-            <span>Date</span>
-          </li>
-        </ul>
-      </>
-    );
-  } else {
-    codefragment = (
-      <>
-        <h1>Reservations Page</h1>
-        <ul className="reservtable">
-          <li className="listitem">
-            <span>Car</span>
-            <span>City</span>
-            <span>Date</span>
-          </li>
-          {reservations.map((reservation) => (
-            <li key={reservation.id} className="listitem">
-              <span>{reservation.item_id}</span>
-              <span>{reservation.city}</span>
-              <span>{reservation.date}</span>
-            </li>
-          ))}
-        </ul>
-      </>
-    );
+// const dispatch = useDispatch();
+
+  // useEffect(() => {
+  //   dispatch(getReserv());
+  // }, [dispatch]);
+
+  const user = useSelector((state) => state.auth.user, shallowEqual);
+  let reservations = useSelector((state) => state.reservations.reservations, shallowEqual);
+  if (user == null || user.name === undefined) {
+    return <Navigate replace to="/login" />;
   }
+  if (reservations === undefined) store.dispatch(getReserv());
+  reservations = reservations || [];
   return (
-    codefragment
+    <>
+      <h1>Reservations Page</h1>
+      <ul className="reservtable">
+        <li className="listitem">
+          <span>Car</span>
+          <span>City</span>
+          <span>Date</span>
+        </li>
+        {reservations.map((reservation) => (
+          <li key={reservation.id} className="listitem">
+            <span>{reservation.item_id}</span>
+            <span>{reservation.city}</span>
+            <span>{reservation.date}</span>
```

Comment:
i think that here we could add the item's name (maybe instead od the id or next to it)

Reply:
It seems easy but it's not, I'm gonna try today